pkg/applier/inventory/inventory_converter_test.go

```diff
@@ -0,0 +1,647 @@
+					Namespace: "obj1-ns",
+					GroupKind: schema.GroupKind{
+						Group: "obj1-group",
+						Kind:  "obj1-kind",
+					},
+				},
+				{
+					Name:      "obj2-name",
+					Namespace: "obj2-ns",
+					GroupKind: schema.GroupKind{
+						Group: "obj2-group",
+						Kind:  "obj2-kind",
+					},
+				},
+			},
+			wantObjStatuses: nil,
+		},
+		"ResourceGroup with spec and status": {
+			fromObj: resourceGroupFactory{
+				spec: v1alpha1.ResourceGroupSpec{
+					Resources: []v1alpha1.ObjMetadata{
+						{
+							Name:      "obj1-name",
+							Namespace: "obj1-ns",
+							GroupKind: v1alpha1.GroupKind{
+								Group: "obj1-group",
+								Kind:  "obj1-kind",
+							},
+						},
+						{
+							Name:      "obj2-name",
+							Namespace: "obj2-ns",
+							GroupKind: v1alpha1.GroupKind{
+								Group: "obj2-group",
+								Kind:  "obj2-kind",
+							},
+						},
+					},
+				},
+				status: v1alpha1.ResourceGroupStatus{
+					ObservedGeneration: int64(invGeneration),
+					ResourceStatuses: []v1alpha1.ResourceStatus{
+						{
+							ObjMetadata: v1alpha1.ObjMetadata{
+								Name:      "obj1-name",
+								Namespace: "obj1-ns",
+								GroupKind: v1alpha1.GroupKind{
+									Group: "obj1-group",
+									Kind:  "obj1-kind",
+								},
+							},
+							Status:     v1alpha1.Current,
+							SourceHash: "abc-123",
+							Strategy:   v1alpha1.Apply,
+							Actuation:  v1alpha1.ActuationSucceeded,
+							Reconcile:  v1alpha1.ReconcileSucceeded,
+						},
+						{
+							ObjMetadata: v1alpha1.ObjMetadata{
+								Name:      "obj2-name",
+								Namespace: "obj2-ns",
+								GroupKind: v1alpha1.GroupKind{
+									Group: "obj2-group",
+									Kind:  "obj2-kind",
+								},
+							},
+							Status:     v1alpha1.Current,
+							SourceHash: "abc-123",
+							Strategy:   v1alpha1.Apply,
+							Actuation:  v1alpha1.ActuationSucceeded,
+							Reconcile:  v1alpha1.ReconcileSucceeded,
+						},
+					},
+				},
+			}.build(t),
+			wantObjRefs: object.ObjMetadataSet{
+				{
+					Name:      "obj1-name",
+					Namespace: "obj1-ns",
+					GroupKind: schema.GroupKind{
+						Group: "obj1-group",
+						Kind:  "obj1-kind",
+					},
+				},
+				{
+					Name:      "obj2-name",
+					Namespace: "obj2-ns",
+					GroupKind: schema.GroupKind{
+						Group: "obj2-group",
+						Kind:  "obj2-kind",
+					},
+				},
+			},
+			// Converter does not currently populate status to Inventory
+			wantObjStatuses: nil,
+		},
+	}
+	for name, tc := range testCases {
+		t.Run(name, func(t *testing.T) {
+			ic := NewInventoryConverter(declared.RootScope, "test-sync", metadata.StatusEnabled)
+			inv, err := ic.InventoryFromUnstructured(tc.fromObj)
+			if tc.wantErr {
+				require.Error(t, err)
+				return
+			}
+			require.NoError(t, err)
+			require.Equal(t, invName, inv.GetName())
+			require.Equal(t, invNamespace, inv.GetNamespace())
+			require.Equal(t, invID, inv.GetID().String())
+			require.Equal(t, tc.wantObjRefs, inv.GetObjectRefs())
```

Comment:
might want testutil.AssertEqual here so it prints a nice diff when it doesn't match.

Reply:
done